diff --git a/sound/soc/qcom/sdm845.c b/sound/soc/qcom/sdm845.c
index cd8b2d43ad3a8..c87d20ecfa22f 100644
--- a/sound/soc/qcom/sdm845.c
+++ b/sound/soc/qcom/sdm845.c
@@ -367,10 +367,30 @@ static int sdm845_snd_startup(struct snd_pcm_substream *substream)
 	case QUATERNARY_MI2S_RX:
 		codec_dai_fmt |= SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_I2S;
-		snd_soc_dai_set_sysclk(cpu_dai,
+		ret = snd_soc_dai_set_sysclk(cpu_dai,
 			Q6AFE_LPASS_CLK_ID_QUAD_MI2S_IBIT,
 			MI2S_BCLK_RATE, SNDRV_PCM_STREAM_PLAYBACK);
-		snd_soc_dai_set_fmt(cpu_dai, fmt);
-		snd_soc_dai_set_fmt(codec_dai, codec_dai_fmt);
+		if (ret < 0) {
+			dev_err(rtd->dev, "QUAT MI2S bit clock error: %d\n",
+				ret);
+			return ret;
+		}
+
+		ret = snd_soc_dai_set_fmt(cpu_dai, fmt);
+		if (ret < 0) {
+			dev_err(rtd->dev, "QUAT MI2S CPU format error: %d\n",
+				ret);
+			return ret;
+		}
+
+		for_each_rtd_codec_dais(rtd, j, codec_dai) {
+			ret = snd_soc_dai_set_fmt(codec_dai, codec_dai_fmt);
+			if (ret < 0) {
+				dev_err(rtd->dev,
+					"QUAT MI2S codec %d fmt error: %d\n",
+					j, ret);
+				return ret;
+			}
+		}
 		break;
 
 	case QUATERNARY_TDM_RX_0:
